maint/local_python/binding_c.py

```diff
@@ -44,25 +44,41 @@ def dump_mpi_c(func, map_type="SMALL"):
             for a in func['include'].replace(',', ' ').split():
                 G.out.append("#include \"%s\"" % a)
 
+#    # Print MPII function prototype
+#    G.out.append("")
+#    func_decl = get_declare_function(func, func['_map_type'])
+#    mpii_decl = re.sub(r'MPI(X?)_([a-zA-Z0-9_]*\()', r'internal\1_\2', func_decl, 1)
+#    G.out.append("/* Use MPIXII prefix for redirection functions only used inside this file */")
+#    dump_line_with_break("static" + mpii_decl + ";")
+
     G.out.append("")
     dump_profiling(func)
 
+    # The MPII (and static) function(s) need to go inside "#ifndef MPICH_MPI_FROM_PMPI" so they don't
+    # get duplicated in the MPI and PMPI versions of the library. Only the MPI-level function should
+    # be in the MPI version of the library.
     if 'polymorph' in func:
         # MPII_ function to support C/Fortran Polymorphism, eg MPI_Comm_get_attr
-        # It needs go inside "#ifndef MPICH_MPI_FROM_PMPI"
-        G.out.pop() # #endif from dump_profiling()
         dump_function(func, kind="polymorph")
-        G.out.append("#endif /* MPICH_MPI_FROM_PMPI */")
+        G.out.append("")
 
-    G.out.append("")
-    dump_manpage(func)
     if 'polymorph' in func:
         dump_function(func, kind="call-polymorph")
     elif 'replace' in func and 'body' not in func:
         dump_function(func, kind="call-replace")
     else:
         dump_function(func, kind="normal")
```

Comment:
It just occurs to me that you can't declare `static` if it's inside `#ifndef MPICH__MPI_FROM_PMPI` block. Because when `MPICH_MPI_FROM_PMPI` is defined, it won't be able to see the `static` symbol that is defined in another unit. I wonder why the `noweak` test didn't capture this.

I think we would prefer using `static`, since it encourages the compiler to inline the function. And if we use `static`, defined it outside the `#ifndef` block. It is okay to have duplicate static function in multiple units, they won't interfere.

Reply:
We only tested `noweak` with `--enable-qmpi`, which only calls `internal_Xxx` from `QMPI_Xxx`, both are only defined in the `pmpi` source. So the tests are ok there. If we test `noweak` with `--disable-qmpi`, it should capture the above mentioned issue.